src/java/org/apache/cassandra/utils/BloomFilter.java

```diff
@@ -90,6 +93,40 @@ private BloomFilter(BloomFilter copy)
         this.bitset = copy.bitset;
     }
 
+    /**
+     * @return true if non-local table's bloom filter should be deserialized on read instead of when opening sstable
```

Comment:
Nit: I'd change `non-local`. Nothing in the patch forbid the usage of lazy loading in pure C*/dse when it will affect sstables that are very local. I think we can just remove that "non-local" but add that this doesn't affect flushed sstables since their BF if not deserialized.